packages/core/lib/debug/printer.js

```diff
@@ -171,13 +171,31 @@ class DebugPrinter {
     const colorizedLines = splitLines(colorizedSource);
 
     this.config.logger.log("");
+    // We clone range though only "lines" is used here.
+    let rangePrintout = JSON.parse(JSON.stringify(range));
+    // We print a warning message and display the end of source code when the
+    // instruction's byte-offset to the start of the range in the source code
+    // is past the end of source code
+    const instruction = this.session.view(solidity.current.instruction);
```

Comment:
I only just noticed something here: This isn't correct!  You're looking at `solidity.current.instruction`, but remember, `location` was passed in.  If the location that was passed in was the current location, this is fine.  But what if the location that was passed in isn't the current location?  You should go by the passed-in location, not the current instruction.